Build fails on CLang 13.0.1

```
[ 45%] Building CXX object src/CMakeFiles/logging_system.dir/logging_system.cpp.o
/home/harrm/projects/soralog/src/logging_system.cpp:173:10: error: no member named 'function' in namespace 'std'
    std::function<int(const std::shared_ptr<const Group> &)> fn =
    ~~~~~^
/home/harrm/projects/soralog/src/logging_system.cpp:173:23: error: expected expression
    std::function<int(const std::shared_ptr<const Group> &)> fn =
                      ^
/home/harrm/projects/soralog/src/logging_system.cpp:173:62: error: use of undeclared identifier 'fn'
    std::function<int(const std::shared_ptr<const Group> &)> fn =
                                                             ^
/home/harrm/projects/soralog/src/logging_system.cpp:187:16: error: use of undeclared identifier 'fn'
      auto n = fn(current->parent());
               ^
/home/harrm/projects/soralog/src/logging_system.cpp:201:16: error: use of undeclared identifier 'fn'
      auto n = fn(it.second);
               ^
/home/harrm/projects/soralog/src/logging_system.cpp:241:10: error: no member named 'function' in namespace 'std'
    std::function<int(const std::shared_ptr<const Group> &)> fn =
    ~~~~~^
/home/harrm/projects/soralog/src/logging_system.cpp:241:23: error: expected expression
    std::function<int(const std::shared_ptr<const Group> &)> fn =
                      ^
/home/harrm/projects/soralog/src/logging_system.cpp:241:62: error: use of undeclared identifier 'fn'
    std::function<int(const std::shared_ptr<const Group> &)> fn =
                                                             ^
/home/harrm/projects/soralog/src/logging_system.cpp:255:16: error: use of undeclared identifier 'fn'
      auto n = fn(current->parent());
               ^
/home/harrm/projects/soralog/src/logging_system.cpp:269:16: error: use of undeclared identifier 'fn'
      auto n = fn(it.second);
               ^
/home/harrm/projects/soralog/src/logging_system.cpp:310:10: error: no member named 'function' in namespace 'std'
    std::function<int(const std::shared_ptr<const Group> &)> fn =
    ~~~~~^
/home/harrm/projects/soralog/src/logging_system.cpp:310:23: error: expected expression
    std::function<int(const std::shared_ptr<const Group> &)> fn =
                      ^
/home/harrm/projects/soralog/src/logging_system.cpp:310:62: error: use of undeclared identifier 'fn'
    std::function<int(const std::shared_ptr<const Group> &)> fn =
                                                             ^
/home/harrm/projects/soralog/src/logging_system.cpp:324:16: error: use of undeclared identifier 'fn'
      auto n = fn(current->parent());
               ^
/home/harrm/projects/soralog/src/logging_system.cpp:338:16: error: use of undeclared identifier 'fn'
      auto n = fn(it.second);
               ^
```

```
$ uname --all
Linux TheFantasyBay 5.17.9-arch1-1 #1 SMP PREEMPT Wed, 18 May 2022 17:30:11 +0000 x86_64 GNU/Linux
```

--- a/src/logging_system.cpp
+++ b/src/logging_system.cpp
@@ -6,6 +6,7 @@
 #include <soralog/logging_system.hpp>
 
 #include <cassert>
+#include <functional>
 #include <iostream>
 #include <set>
 

